aws/resource_aws_ebs_snapshot_import.go

```diff
@@ -0,0 +1,427 @@
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"comment": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"upload_end": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+						"upload_size": {
+							Type:     schema.TypeFloat,
+							Optional: true,
+							ForceNew: true,
+						},
+						"upload_start": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+					},
+				},
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"disk_container": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"format": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								ec2.DiskImageFormatVmdk,
+								ec2.DiskImageFormatVhd,
+							}, false),
+						},
+						"url": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ExactlyOneOf: []string{"disk_container.0.user_bucket", "disk_container.0.url"},
+						},
+						"user_bucket": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							ForceNew:     true,
+							ExactlyOneOf: []string{"disk_container.0.user_bucket", "disk_container.0.url"},
+							MaxItems:     1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"s3_bucket": {
+										Type:     schema.TypeString,
+										Required: true,
+										ForceNew: true,
+									},
+									"s3_key": {
+										Type:     schema.TypeString,
+										Required: true,
+										ForceNew: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"owner_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner_alias": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"encrypted": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"volume_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"role_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"data_encryption_key_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsEbsSnapshotImportCreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
It seems this cannot create a resource with tags, but it can add tags later. Todo: look in to this.

```
Error: error importing EBS Snapshot: InvalidParameter: 'snapshot' is not a valid taggable resource type for this operation.
	status code: 400, request id: d49f7fd0-4c06-48bb-a976-8ea12b9ed8ef
```